splitter/processor.go

```diff
@@ -86,9 +97,41 @@ func (p *Processor) Process(r io.Reader, event *event.FileUploaded, startTime ti
 	})
 }
 
+func sendDatasetSplitEvent(datasetID string, totalRows int) {
+
+	message := DatasetSplitEvent{
+		DatasetID: datasetID,
+		TotalRows: totalRows,
+		SplitTime: time.Now().UTC().Unix() * 1000, // unix time in milliseconds
+	}
+
+	messageJSON, err := json.Marshal(message)
+	if err != nil {
+		log.Error(err, log.Data{
+			"details": "Could not create the json representation of message",
+			"message": messageJSON,
+		})
+		panic(err)
```

Comment:
I'm going to leave the panic, because if we can't marshal the message into json there's something fundamentally wrong with the code. That message is internally generated so should never fail to be marshaled.